plugins/modules/fusion_se.py

```diff
@@ -170,51 +178,32 @@ def get_se(module, fusion):
 
 def create_se(module, fusion):
     """Create Storage Endpoint"""
-
     se_api_instance = purefusion.StorageEndpointsApi(fusion)
 
-    changed = True
-
     if not module.check_mode:
-        if not module.params["display_name"]:
-            display_name = module.params["name"]
-        else:
-            display_name = module.params["display_name"]
-        ifaces = []
-        for address in module.params["addresses"]:
-            if module.params["gateway"]:
-                iface = purefusion.StorageEndpointIscsiDiscoveryInterfacePost(
-                    address=address,
-                    gateway=module.params["gateway"],
-                    network_interface_groups=module.params["network_interface_groups"],
-                )
-            else:
-                iface = purefusion.StorageEndpointIscsiDiscoveryInterfacePost(
-                    address=address,
-                    network_interface_groups=module.params["network_interface_groups"],
-                )
-            ifaces.append(iface)
-        op = purefusion.StorageEndpointPost(
+        storage_endpoint = purefusion.StorageEndpointPost(
+            name=module.params["name"],
+            display_name=module.params["display_name"] or module.params["name"],
             endpoint_type=module.params["endpoint_type"],
             iscsi=purefusion.StorageEndpointIscsiPost(
-                discovery_interfaces=ifaces,
+                discovery_interfaces=[
```

Comment:
Took me way too long to recognize this is a list comprehension 😄 Maybe it would be better to refactor it out into its own few statements to make sure it fits into one line so it is not so confusing

Reply:
What about this?
```
def create_se(module, fusion):
    """Create Storage Endpoint"""
    se_api_instance = purefusion.StorageEndpointsApi(fusion)

    if not module.check_mode:
        discovery_interfaces = [
            purefusion.StorageEndpointIscsiDiscoveryInterfacePost(**endpoint)
            for endpoint in module.params["iscsi"]
        ]
        iscsi = purefusion.StorageEndpointIscsiPost(
            discovery_interfaces=discovery_interfaces
        )
        storage_endpoint = purefusion.StorageEndpointPost(
            name=module.params["name"],
            display_name=module.params["display_name"] or module.params["name"],
            endpoint_type=module.params["endpoint_type"],
            iscsi=iscsi,
        )
        op = se_api_instance.create_storage_endpoint(
            storage_endpoint,
            module.params["region"],
            module.params["availability_zone"],
        )
        await_operation(module, fusion, op)

    module.exit_json(changed=True)
```